census/vanilla-tf/trainer/task.py

```diff
@@ -0,0 +1,335 @@
+
+  # convert the continuous columns into tf.constant tensor
+  continuous_columns = [
+      tf.constant(input_df[col].values) for col in CONTINUOUS_COLS
+  ]
+
+  # convert the categorical columns into sparse tensors
+  categorical_columns = [
+      tf.SparseTensor(
+          indices=[[i, 0] for i in range(input_df[col].size)],
+          values=input_df[col].astype('category').cat.codes.values,
+          dense_shape=[input_df[col].size, 1])
+      for col in CATEGORICAL_COLS
+  ]
+
+  # convert the labels into one hot encoding
+  label_tensor = tf.one_hot(
+      label_df.astype('category').cat.codes.values,
+      2, off_value=1, on_value=0)
+
+  return continuous_columns + categorical_columns, label_tensor
+
+
+#
+# Feature crosses and generation of wide columns
+#
+def sparse_cross(feature_tensors, num_buckets, name='cross'):
+  """Sparse feature cross of the feature SparseTensors."""
+  return sparse_feature_cross_op.sparse_feature_cross(
+      feature_tensors,
+      hashed_output=True,
+      num_buckets=num_buckets,
+      hash_key=tf.contrib.layers.SPARSE_FEATURE_CROSS_DEFAULT_HASH_KEY,
+      name=name
+  )
+
+def generate_wide_columns(input_columns):
+  """Generate wide columns by adding feature crosses of SparseTensors."""
+  (age, education_num, capital_gain, capital_loss, hours_per_week,
+   gender, race, education, marital_status, relationship, workclass,
+   occupation, native_country) = input_columns
+
+  # bucketize the age feature
+  # See https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/ops/bucketization_op.py
+  age_bucket = bucketization_op.bucketize(age, [18, 25, 30, 35, 40, 45, 50, 55, 60, 65])
+
+  wide_columns = [
+      (sparse_cross([education, occupation], 15 * 16, name='edu_occ'), 15 * 16),
+      (gender, 2),
+      (native_country, 42),
+      (education, 16),
+      (occupation, 15),
+      (workclass, 9),
+      (marital_status, 7),
+      (relationship, 6)
+  ]
+
+  dense_tensors = [sparse_to_dense(col[0], col[1]) for col in wide_columns]
+
+  return tf.concat(dense_tensors, 1)
+
+def sparse_to_dense(sparse_tensor, vocab_size):
+  """Convert the sparse to dense tensor."""
+  dense_tensor = tf.sparse_to_indicator(sparse_tensor, vocab_size)
+  dense_tensor = tf.cast(dense_tensor, tf.int32)
+  return dense_tensor
+
+def read_input_tensor(input_file, skiprows=None):
+  """Concatenate the wide columns to produce a single tensor."""
+  inp, label = read_input_data(input_file, skiprows)
+  in_tensor, label_tensor = generate_input(inp, label)
+  return generate_wide_columns(in_tensor), label_tensor
+
+
+#
+# Function to perform the actual training loop.
+# This function is same for single and distributed.
+#
+def training_steps(session, train_step, eval_step, logits,
+                   labels, global_step, max_steps, inp, label,
+                   eval_inp, eval_label, job_name='local', job_id=0):
+  """Run the training steps and calculate accuracy every 10 steps."""
```

Comment:
"every 10 steps" Because there is not batching, this is every 10 epochs, right? 

Maybe expand comment saying something like:
Calculate accuracy every 10 steps, because each train set is over the whole dataset, the calculation runs every 10 epochs. Note that for larger datasets, it is not advised to run evaluation so frequently.